applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1693,6 +1777,99 @@ void BaseSolidElement::CalculateAndAddExtForceContribution(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void BaseSolidElement::CalculateLumpedMassVector(VectorType& rMassVector)
+{
+    KRATOS_TRY;
+
+    const auto& r_geom = GetGeometry();
+    const auto& r_prop = GetProperties();
+    const SizeType dimension = r_geom.WorkingSpaceDimension();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType mat_size = dimension * number_of_nodes;
+
+    // Clear matrix
+    if (rMassVector.size() != mat_size)
+        rMassVector.resize( mat_size, mat_size );
+
+    const double density = r_prop[DENSITY];
+    const double thickness = (dimension == 2 && r_prop.Has(THICKNESS)) ? r_prop[THICKNESS] : 1.0;
+
+    // LUMPED MASS MATRIX
+    const double total_mass = GetGeometry().DomainSize() * density * thickness;
+
+    Vector lumping_factors;
+    lumping_factors = GetGeometry().LumpingFactors( lumping_factors );
```

Comment:
This is correct, however i am not too convinced the lumping factors are correctly implemented for high order elements (To be verified &| corrected in a different PR).

Actually i think it is even wrong for quad elements since in the case the shape is skewed it is not 0.25 0.25 0.25 0.25

the fast way to compute it (which we could implement in the geometry base class) is something like

      lumping_factors.clear()
      loop on gauss points
           auto N = row(NMatrix, gauss)
           lumping_factors += gauss.weight * detJ * N

(note that i am NOT doing  outer_prod(N,N) but only using N once)

This loop can be done with the default integration rule, (which is one order lower than the one used for integrating the consistent version)

as a final note beware that from this mass lumping technique you will obtain negative values of the mass for example on the corner points of quadratic triangles. As i recall this technique also gives zero on corners of 9 noded quads

Reply:
https://scicomp.stackexchange.com/questions/19704/how-to-formulate-lumped-mass-matrix-in-fem

Reply:
> (To be verified &| corrected in a different PR).

:D for using bitwise-syntax to explain sth @RiccardoRossi

Reply:
@loumalouomega yes, the technique you point to makes sense. actually i would advocate for adding a method to the geometry for computing this (removing the LumpingFactors which is a halfway step in the correct direction). That would simplify the elements, particularly in the case of using the consistent version. No doubts this is a different PR...just consider it as an idea here

          Vector M = geom.LumpedMassMatrix()
          Matrix M = geom.ConsistentMassMatrix()